delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/operations/disk/DiskStorageFetchFile.scala

```diff
@@ -20,7 +20,9 @@ object DiskStorageFetchFile extends FetchFile {
 
   override def apply(path: Uri.Path): IO[FetchFileRejection, AkkaSource] =
     Try(Paths.get(URI.create(s"file://$path"))) match {
-      case Failure(err)   => IO.raiseError(UnexpectedLocationFormat(s"file://$path", err.getMessage))
-      case Success(value) => UIO.pure(FileIO.fromPath(value))
+      case Failure(err)  => IO.raiseError(UnexpectedLocationFormat(s"file://$path", err.getMessage))
+      case Success(path) =>
+        IO.raiseWhen(!path.toFile.exists())(FetchFileRejection.FileNotFound(path.toString))
```

Comment:
Would want to double check this is how we interact with the file system, I wonder if there is a different library we might use. FileIO didn't seem to have this but perhaps I missed it